docs/content/guides/dagster/migrating-to-pythonic-resources-and-config.mdx

```diff
@@ -0,0 +1,295 @@
+def asset_one(context: OpExecutionContext) -> None:
+    # this still works because the resource is still available on the context
+    assert context.resources.fancy_db
+
+defs = Definitions(
+    assets=[asset_one],
+    resources={"fancy_db": FancyDbResource(conn_string="some_value")},
+)
+```
+
+Now we can move the asset to take the resource as a parameter.
+
+```python file=/guides/dagster/migrating_to_python_resources_and_config/migrating_resources.py  startafter=begin_new_style_resource_on_param endbefore=end_new_style_resource_on_param dedent=4
+from dagster import OpExecutionContext, asset
+
+@asset
+def asset_one(context: OpExecutionContext, fancy_db: FancyDbResource) -> None:
+    assert fancy_db
+```
+
+## Migrating a resource that uses a separate object for business logic
+
+Frequently users want to separate out the interaction with the configuration system from the object that contains the actual business logic. Maybe it is pre-existing or third-party code that knows nothing about the configuration system and shouldn't. Or it is a complex client that requires mutable state and bookkeeping. In the old API the engineer would have written a `@resource` decorator that returns that object directly.
+
+```python file=/guides/dagster/migrating_to_python_resources_and_config/migrating_resources.py  startafter=begin_old_third_party_resource endbefore=end_old_third_party_resource dedent=4
+# Pre-existing code that you don't want to alter
+class FancyDbClient:
+    def __init__(self, conn_string: str) -> None:
+        self.conn_string = conn_string
+
+    def execute_query(self, query: str) -> None:
+        ...
+
+from dagster import InitResourceContext, asset, resource
+
+@resource(config_schema={"conn_string": str})
+def fancy_db_resource(context: InitResourceContext) -> FancyDbClient:
+    return FancyDbClient(context.resource_config["conn_string"])
+
+@asset(required_resource_keys={"fancy_db"})
+def existing_asset(context) -> None:
+    context.resources.fancy_db.execute_query("SELECT * FROM foo")
+```
+
+With Pythonic style resources you would write a class that can return that client from a method. In code that consumes that resource you would call `get_client()` access the underlying client.
+
+```python file=/guides/dagster/migrating_to_python_resources_and_config/migrating_resources.py  startafter=begin_new_third_party_resource endbefore=end_new_third_party_resource dedent=4
+from dagster import ConfigurableResource, asset
+
+class FancyDbResource(ConfigurableResource):
+    conn_string: str
+
+    def get_client(self) -> FancyDbClient:
+        return FancyDbClient(self.conn_string)
+
+@asset
+def new_asset(fancy_db: FancyDbResource) -> None:
+    client = fancy_db.get_client()
+    client.execute_query("SELECT * FROM foo")
+```
+
+`@resource` functions could also be context managers to handle initialization and cleanup tasks. This context manager was called managed by framework code rather than the user's code.
```

Comment:
> This context manager was called managed by framework code

"called managed" reads odd to me here - drop one of the verbs?

Reply:
most def. typo